components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandler.java

```diff
@@ -350,7 +350,8 @@ protected void handlePostAuthentication(HttpServletRequest request,
                         }
                         authenticatedUserAttributes = FrameworkUtils.buildClaimMappings(mappedAttrs);
                         if (isSPStandardClaimDialect(context.getRequestType(), appConfig) &&
-                                authenticatedUserAttributes.isEmpty()) {
+                                authenticatedUserAttributes.isEmpty() &&
+                                sequenceConfig.getAuthenticatedUser() != null) {
```

Comment:
This is configured as the subject attribute step. How can the AuthenticatedUser be null? will it working the subject attribute correctly? 

@piraveena

Reply:
In the tested scenario, subject identifier is selected from 2nd step(Basic authenticator) and 1st step(Federated authenticator) is selected to get the user attributes. when the step configs are iterating it will try to get the authenticated user before the authenticated user set in the subject identifier step. That's why the authenticated user be null here.